### PR TITLE
[XLA:GPU] Update cuDNN flash attention backward to use new cuDNN frontend (>1.0)

### PR DESCRIPTION
**Pull request by Cjkkkk**
* Replace complicated node by node flash attention cuDNN graph construction with simple cudnn sdpa call provided in new cuDNN frontend (>1.0).
* Remove softmax_sum/dq_accum tensors from the custom call outputs since it is not required anymore in new cuDNN frontend.
* Add a pass to query workspace from cuDNN graph instead of manual calculation as workspace size is subject to change.

### CODE REVIEW

File: xla/stream_executor/cuda/cuda_dnn.cc [hunk 1]
```diff
@@ -10425,6 +9760,7 @@ absl::Status CudnnGraph::Build(dnn::DnnSupport& dnn_support,
   TF_ASSIGN_OR_RETURN(auto cudnn, cudnn_support.cudnn_->GetLocalHandle());
   RETURN_IF_CUDNN_FRONTEND_ERROR(
       graph_.build_plan_at_index(cudnn->handle(), plan_id));
+  // RETURN_IF_CUDNN_FRONTEND_ERROR(graph_.build_plans(cudnn->handle()));
```
Comment (akuegel) [inline xla/stream_executor/cuda/cuda_dnn.cc:9763]: I guess this line can be removed?

Comment (Cjkkkk) [reply xla/stream_executor/cuda/cuda_dnn.cc]: fixed.

**Review by akuegel — APPROVED** [review]

**Comment by akuegel** [comment]
It looks like there is a missing dependency:

error: module //third_party/tensorflow/compiler/xla/service/gpu:cudnn_workspace_rewriter does not depend on a module exporting 'third_party/tensorflow/compiler/xla/service/gpu/cublas_cudnn.h'

Can you please add that?

**Review by akuegel — APPROVED** [review]

**Comment by Cjkkkk** [comment]
@akuegel the google internal test failure should be fixed now.

**Review by akuegel — APPROVED** [review]

**Comment by akuegel** [comment]
@Cjkkkk I am currently in the process of moving HLO passes to the transforms subdirectory. In this process I noticed that we are missing a test for the CudnnWorkspaceRewriter pass which was added in this PR (I missed this during the review). Can you please write a test?

**Pull request closed without merging**